### PR TITLE
Bugfix/um 703 mpi for ci

### PR DESCRIPTION
**Pull request by mcfadden8**

### CODE REVIEW

**Comment by mcfadden8** [comment]
LGTM

File: cmake/UmpireMacros.cmake [hunk 1]
```diff
@@ -27,3 +27,31 @@ macro(umpire_add_code_checks)
...
+    set(options)
+    set(singleValueArgs NAME NUM_MPI_TASKS)
+    set(multiValueArgs COMMAND)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+     "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+   if (ENABLE_MPI)
+      # The 'CONFIGURATIONS Benchmark' line excludes benchmarks
```
Comment (davidbeckingsale) [inline cmake/UmpireMacros.cmake:47]: I don't think this comment should be here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to
 
 - Added a getParent functionality for retrieving the memory resource of an allocator.
 
+- Added gitlab testing of Umpire when MPI is enabled
```
Comment (davidbeckingsale) [inline CHANGELOG.md:21]:
```suggestion
- Added GitLab testing of Umpire when MPI is enabled
```

File: tests/integration/io/CMakeLists.txt [hunk 3]
```diff
@@ -12,14 +12,32 @@ blt_add_executable(
...
-  # Temporarily disable this test when MPI is enabled.  This will be
-  # re-enabled again for MPI when UM-703 has been fixed.
-  #
-  if (PYTHON_EXECUTABLE AND NOT ENABLE_MPI)
-    add_test(
-      NAME io_tests
-      COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/io_tests_runner.py WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+  if (PYTHON_EXECUTABLE)
+    umpire_add_test_with_mpi(
+      NAME io_tests_1
```
Comment (davidbeckingsale) [inline tests/integration/io/CMakeLists.txt:17]: Can the names be more descriptive than _1, _2 please?

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
Couple of quick changes. Also, is the radiuss-spack-config submodule pointing at a branch?

**Comment by mcfadden8** [comment]
LGTM

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
Can you point the submodule at the latest "main" - the PR with MPI configs was merged.

**Review by davidbeckingsale — APPROVED** [review]

**Comment by mcfadden8** [comment]
LGTM

**Pull request merged**